lumen/ai/agents.py

```diff
@@ -69,24 +69,17 @@ class Agent(Viewer):
     __abstract = True
 
     def __init__(self, **params):
-        self._retries_left = 1
         def _exception_handler(exception):
-            import traceback
-            tb = traceback.format_exc()
-            print(tb)
-            if self._retries_left > 0:
-                self._retries_left -= 1
-                self.interface.send(
-                    f"Taking a different approach to expertly resolve this issue `{exception}` using world-class knowledge.",
-                    user="Exception",
-                )
-            else:
-                self.interface.send(
-                    f"Error cannot be resolved: `{exception}`.",
-                    user="System",
-                    respond=False
-                )
+            if str(exception) in self.interface.serialize()[-1]["content"]:
```

Comment:
Prevent redundant message sent twice.

Reply:
Can you not just serialize the last message?